apps/condo/domains/ticket/components/ShareTicketModal.tsx

```diff
@@ -0,0 +1,328 @@
+import { Button } from '@condo/domains/common/components/Button'
+import { green } from '@ant-design/colors'
+import Link from 'next/link'
+import { useIntl } from '@core/next/intl'
+import { useMutation } from '@core/next/apollo'
+import { SHARE_TICKET_MUTATION } from '@condo/domains/ticket/gql'
+import { useOrganization } from '@core/next/organization'
+import { getEmployeeWithEmail } from '@condo/domains/ticket/utils/clientSchema/search'
+import { GraphQlSearchInput } from '@condo/domains/common/components/GraphQlSearchInput'
+import styled from '@emotion/styled'
+import { colors } from '@condo/domains/common/constants/style'
+import { getClientSideSenderInfo } from '@condo/domains/common/utils/userid.utils'
+import { useRouter } from 'next/router'
+
+const collapse = css`
+    border-radius: 8px;
+    background: ${colors.white};
+
+    & .ant-collapse-content {
+        border-top: none;
+
+    }
+
+    & .ant-collapse-item:last-child,
+    & .ant-collapse-item:last-child > .ant-collapse-header {
+        border-radius: 8px;
+        line-height: 24px
+    }
+
+    & > .ant-collapse-item > .ant-collapse-header .ant-collapse-arrow {
+        right: 20px;
+        font-size: 14px;
+    }
+    & > .ant-collapse-item > .ant-collapse-header {
+        color: ${green[6]};
+        padding: 17px 20px;
+        font-size: 16px;
+
+        &:hover, &:focus {
+            color: ${green[5]};
+        }
+    }
+    & .ant-collapse-item:last-child > .ant-collapse-content {
+        border-radius: 8px;
+    }
+    & .ant-collapse-content > .ant-collapse-content-box {
+        padding: 0px 19px 22px;
+    }
+
+    & .ant-select-multiple .ant-select-selection-item {
+        background: ${colors.white};
+        font-size: 12px;
+    }
+
+    & .ant-select-multiple.ant-select-lg .ant-select-selection-item {
+        line-height: 20px;
+        height: 20px;
+        border: none;
+    }
+
+    & .ant-select-item-option-content,
+    & .ant-select-item,
+    & .ant-select-show-search.ant-select:not(.ant-select-customize-input) .ant-select-selector input {
+        font-size: 12px;
+    }
+
+`
+
+const search = css`
+    width: 100%;
+`
+
+const ModalHeader = styled.span`
+    font-weight: 700;
+    line-height: 32px;
+    font-size: 24px;
+`
+
+const ShareButton = styled.span`
+  border: 1px solid ${colors.lightGrey[5]};
+  border-radius: 8px;
+  display: inline-block;
+  width: 100%;
+  color: ${green[6]};
+  cursor: pointer;
+  height: 60px;
+  line-height: 60px;
+  position: relative;
+  padding-left: 20px;
+  font-size: 16px;
+
+  &:hover, &:focus {
+      color: ${green[5]};
+  }
+
+  & .anticon {
+    position: absolute;
+    right: 20px;
+    top: 50%;
+    transform: translateY(-50%);
+    font-size: 14px;
+    margin-top: 1px;
+  }
+`
+
+const declOfNum = (number, words) => words[(number % 100 > 4 && number % 100 < 20) ? 2 : [2, 0, 1, 1, 1, 2][(number % 10 < 5) ? Math.abs(number) % 10 : 5]]
```

Comment:
Better to avoid such situations as it will be hard to implement it for different languages
And if you decide to use such functions - you need to move this to utils and add some tests